hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -232,14 +245,19 @@ public Vertex onFullScan(FullScanPhysicalRel rel) {
 
         dagBuildContext.setTable(table);
         dagBuildContext.setRel(rel);
-        return getJetSqlConnector(table).fullScanReader(
+        SqlConnector sqlConnector = getJetSqlConnector(table);
+        List<List<Expression<?>>> mapPartitionsToScanExprs = null;
+        if (sqlConnector.equals(IMapSqlConnector.INSTANCE)) {
+            mapPartitionsToScanExprs = computeRequiredPartitionsToScan(table.getSqlName());
+        }
+        return sqlConnector.fullScanReader(
                 dagBuildContext,
                 wrap(rel.filter()),
                 wrap(rel.projection()),
+                mapPartitionsToScanExprs,
```

Comment:
can't you pass `prunabilities.get(mapName);` and move rest of `computeRequiredPartitionsToScan` to IMap connector?

Reply:
Good catch btw, ty.

Reply:
https://github.com/hazelcast/hazelcast/pull/25006/commits/71cfe45ba4334b96de3a4e540840d6efd81f937d